sensor/kubernetes/listener/resources/selector.go

```diff
@@ -1,45 +1,118 @@
 package resources
 
 import (
+	"math"
+
 	"k8s.io/apimachinery/pkg/labels"
 )
 
-// selector is a restricted version of labels.Selector
+// labelWithLen is label.Labels with added Len() function
+type labelsWithLen interface {
+	Has(label string) (exists bool)
+	Get(label string) (value string)
+	Len() uint
+}
+
+// selector is a restricted version of selectorWrapper
 type selector interface {
+	Matches(labelsWithLen) bool
+}
+
+// internalSelector is a restricted version of labels.Selector
+type internalSelector interface {
 	Matches(labels.Labels) bool
 }
```

Comment:
I am really confused by so many interfaces and types: selector, internalSelector, restrictedSelector, selectorWrapper.

Could we try to simplify this?

Some ideas how could we do that:
1. Maybe we should use only `labelsWithLen` in our entire code base - would that be possible?
2. Maybe `labelsWithLen` could have a single implementation `labelsWithLenImpl` (or similar name) - would that be possible?

Reply:
I am happy to help more in case you would need that, but I don't want to steal you the fun of refactoring this :)

Reply:
Perhaps I overused interfaces this time around. I think I misunderstood your comment about defining more interfaces consumer side even if they're duplicates to mean that we want a lot more interfaces 😅 I have no problem refactoring it to remove a bunch of them and change my `labelWrapper` to `labelsWithLenImpls`